### ISSUE TITLE
Creating DateTimes.fromObject with fractional values results in NaN

### ISSUE THREAD

**Issue by ihodes** [moment/luxon#556]
I came across an odd edge-case today which was a lot of fun to debug. (Would be nice if we could have an `int` type in TS)

Though using `numbers` in `fromObject` typically works, when you use 59.X it results in NaN's in the resulting object. `minute: 30.5` and `minute: 59` works, but `minute: 59.1` does not. 

Not sure if you really want to introduce a runtime check for this issue, though, but worth reporting (couldn't find a similar report).

```
const d = DateTime.fromObject({
      day: date.day,
      month: date.month,
      year: date.year,
      hour: 11,
      minute: 59.1
})
d.hour => NaN
d.minute => NaN
```

**Comment by GillesDebunne** [comment]
Thanks a lot for reporting, this definitively looks like a bug.

**Comment by icambron** [comment]
To be fair to the resulting object, it isn't _just_ a bunch of `NaN`s; it's an explicitly invalid DateTime:

```js
d.invalidReason //=> "you specified 59.1 (of type number) as a minute, which is invalid"
```

[Invalid DateTimes](https://moment.github.io/luxon/docs/manual/validity.html) are a bit painful and in 2.0 we're just going to throw an error instead. But in any case, it's fair to ask what _should_ happen here, because I certainly agree that 59.1 being different from 35.2 is bad, which currently happens because Luxon is applying the rule "must be a number between 1 and 59".

There are a few different behaviors we could have:

1. Reject non-integers entirely, so that they're all the same. That would currently be an invalid result, 
2. Round the number
3. Pass it through to the underlying Date. This has the effect, at least in v8, of rounding the number down
4. Do the math of turning decimal minutes into the right number of minutes, seconds, and rounded milliseconds

2 and 3 are silent data loss, so those are nonstarters. 4 is tempting, but also a lot of work. 1 is consistent with the TypeScript change we are making in 2.0 anyway, and also simple and easy to enforce. So 1 is the winner.

Fixed 87d6ae2b and released in 1.19.0

**Comment by ihodes** [comment]
Thanks for writing out the reasoning, very nice thing to do. I completely agree with the resolution, and just wish TS had an integer type…
